trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/standardfunctions/StandardFunctionHandler.java

```diff
@@ -2407,18 +2460,56 @@ private Result handleMemCopyOrMove(final IDispatcher main, final IASTFunctionCal
 	private Result handleErrorFunction(final IDispatcher main, final IASTFunctionCallExpression node,
 			final ILocation loc, final String name) {
 		final Expression falseLiteral = ExpressionFactory.createBooleanLiteral(loc, false);
-		final Statement st =
-				createReachabilityAssert(loc, name, mSettings.checkErrorFunction(), Spec.ERROR_FUNCTION, falseLiteral);
+		final Statement st = createReachabilityAssert(loc, name, mSettings.checkErrorFunction(),
+				ISpec.Type.ERROR_FUNCTION, falseLiteral);
 		return new ExpressionResult(Collections.singletonList(st), null);
 	}
 
 	/**
-	 * Create "assert expr" or "assume expr" for usage in reachability specifications, depending on the settings. If
-	 * checkProperty is true (i.e. the check is enabled), an "assert expr" will be generated, otherwise an "assume expr"
-	 * will be generated.
+	 * Create an assertion or assumption for usage in reachability specifications.
+	 * 
+	 * @param loc
+	 *            location of the reachability specification.
+	 * @param functionName
+	 *            name of the function for the reachability specification.
+	 * @param checkProperty
+	 *            enables checking of the reachability specification, otherwise an assumption is made.
+	 * @param spec
+	 *            type of the reachability specification that should be checked.
+	 * @param expr
+	 *            expression for checking the reachability specification.
+	 * 
+	 * @see {@link #createReachabilityAssert(ILocation, String, boolean, Spec, Expression, String)}
+	 */
+	private Statement createReachabilityAssert(final ILocation loc, final String functionName,

```

Comment:
I think the comment is somewhat misleading. The function creates an assume statement or an assert statement annotated with a check annotation. It is somewhat strange that it is called "createReachabilityAssert". Also, a "reachability specification" is a) not a really well defined term, and b) just not necessary here. 
Then, the parameter function name is only relevant for certain types of Specs, so the comment is probably also misleading.

Reply:
You're right. The function was already named this way before my changes. It's probably better if we refactor it all ...